### PR TITLE
updated references to the download page to instead reference latest.m…

### PR DESCRIPTION
**Pull request by ztrayner**
#### Summary
Updated references to the download page to instead reference latest.mattermost.com and the version archive.

### CODE REVIEW

File: source/install/install-debian-mattermost.rst [hunk 1]
```diff
@@ -11,13 +11,13 @@ Assume that the IP address of this server is 10.10.10.2.
 
 1. Log in to the server that will host Mattermost Server and open a terminal window.
 
-2. Download `the latest version of the Mattermost Server <https://about.mattermost.com/download/>`__. In the following command, replace ``X.X.X`` with the version that you want to download:
+2. Download the latest version of the Mattermost Server. To view other release versions, release notes, and checksums visit `the version archive <../administration/version-archive.html>`__.
 
-  ``wget https://releases.mattermost.com/X.X.X/mattermost-X.X.X-linux-amd64.tar.gz``
+  ``wget http://latest.mattermost.com/server-linux-amd64 -O mattermost-server``
```
Comment (jaydeland) [inline source/install/install-debian-mattermost.rst]: Should we mention both Enterprise and Team?
http://latest.mattermost.com/server-linux-team
http://latest.mattermost.com/server-linux-enterprise

File: source/administration/upgrade.rst [hunk 2]
```diff
@@ -38,25 +38,25 @@ Location of your local storage directory
...
 
-#. Download `the latest version of Mattermost Server <https://about.mattermost.com/download/>`__. In the following command, replace ``X.X.X`` with the version that you want to download:
+#. Download the latest version of the Mattermost Server. To view other release versions, release notes, and checksums visit `the version archive <./version-archive.html>`__.
 
    *Enterprise Edition*
 
    .. code-block:: sh
 
-     wget https://releases.mattermost.com/X.X.X/mattermost-X.X.X-linux-amd64.tar.gz
+     wget http://latest.mattermost.com/server-linux-amd64 -O mattermost-server
```
Comment (jaydeland) [inline source/administration/upgrade.rst]: http://latest.mattermost.com/server-linux-enterprise

File: source/administration/upgrade.rst [hunk 3]
```diff
@@ -38,25 +38,25 @@ Location of your local storage directory
...
 
    *Team Edition*
 
    .. code-block:: sh
 
-    wget https://releases.mattermost.com/X.X.X/mattermost-team-X.X.X-linux-amd64.tar.gz
+    wget http://latest.mattermost.com/server-linux-amd64-team -O mattermost-server
```
Comment (jaydeland) [inline source/administration/upgrade.rst]: http://latest.mattermost.com/server-linux-team

File: source/install/install-rhel-6-mattermost.rst [hunk 4]
```diff
@@ -11,13 +11,13 @@ Assume that the IP address of this server is 10.10.10.2
 
 1. Log in to the server that will host Mattermost Server and open a terminal window.
 
-2. Download `the latest version of the Mattermost Server <https://about.mattermost.com/download/>`__. In the following command, replace ``X.X.X`` with the version that you want to download:
+2. Download the latest version of the Mattermost Server. To view other release versions, release notes, and checksums visit `the version archive <../administration/version-archive.html>`__.
 
-  ``wget https://releases.mattermost.com/X.X.X/mattermost-X.X.X-linux-amd64.tar.gz``
+  ``wget http://latest.mattermost.com/server-linux-amd64 -O mattermost-server``
```
Comment (jaydeland) [inline source/install/install-rhel-6-mattermost.rst]: Should we mention both Enterprise and Team?
http://latest.mattermost.com/server-linux-team
http://latest.mattermost.com/server-linux-enterprise

File: source/install/install-rhel-7-mattermost.rst [hunk 5]
```diff
@@ -11,13 +11,13 @@ Assume that the IP address of this server is 10.10.10.2
 
 1. Log in to the server that will host Mattermost Server and open a terminal window.
 
-2. Download `the latest version of the Mattermost Server <https://about.mattermost.com/download/>`__. In the following command, replace ``X.X.X`` with the version that you want to download:
+2. Download the latest version of the Mattermost Server. To view other release versions, release notes, and checksums visit `the version archive <../administration/version-archive.html>`__.
 
-  ``wget https://releases.mattermost.com/X.X.X/mattermost-X.X.X-linux-amd64.tar.gz``
+  ``wget http://latest.mattermost.com/server-linux-amd64 -O mattermost-server``
```
Comment (jaydeland) [inline source/install/install-rhel-7-mattermost.rst]: Should we mention both Enterprise and Team?
http://latest.mattermost.com/server-linux-team
http://latest.mattermost.com/server-linux-enterprise

File: source/install/install-ubuntu-1604-mattermost.rst [hunk 6]
```diff
@@ -11,13 +11,13 @@ Assume that the IP address of this server is 10.10.10.2.
 
 1. Log in to the server that will host Mattermost Server and open a terminal window.
 
-2. Download `the latest version of the Mattermost Server <https://about.mattermost.com/download/>`__. In the following command, replace ``X.X.X`` with the version that you want to download:
+2. Download the latest version of the Mattermost Server. To view other release versions, release notes, and checksums visit `the version archive <../administration/version-archive.html>`__.
 
-  ``wget https://releases.mattermost.com/X.X.X/mattermost-X.X.X-linux-amd64.tar.gz``
+  ``wget http://latest.mattermost.com/server-linux-amd64 -O mattermost-server``
```
Comment (jaydeland) [inline source/install/install-ubuntu-1604-mattermost.rst]: Should we mention both Enterprise and Team?
http://latest.mattermost.com/server-linux-team
http://latest.mattermost.com/server-linux-enterprise

File: source/administration/upgrade.rst [hunk 7]
```diff
@@ -38,25 +38,28 @@ Location of your local storage directory
 
      cd /tmp
 
-#. Download `the latest version of Mattermost Server <https://about.mattermost.com/download/>`__. In the following command, replace ``X.X.X`` with the version that you want to download:
+#. Download the latest version of the Mattermost Server. To view other release versions, release notes, and checksums visit `the version archive <./version-archive.html>`__.
```
Comment (amyblais) [inline source/administration/upgrade.rst]:
```suggestion
#. Download the latest version of the Mattermost Server. To view other release versions, release notes, and checksums visit `the version archive <../administration/version-archive.html>`__.

```

**Comment by amyblais** [comment]
@ztrayner Following up on Carlos's comment, is there a particular reason why we want to reference latest.mattermost.com?

**Comment by ztrayner** [comment]
@amyblais @cpanato - The reason is because currently our [marketing download page](https://mattermost.com/download/) sends people to the docs for installation instructions. Then the docs currently sends people back to the marketing download page to grab the string of the server download.  It's a weird circular dependency.  By having a URL that provides the latest download we don't need to send people to an external page just to get the download URL string.

@jaydeland worked on creating the endpoint at latest.mattermost.com.  I am indifferent to the format of the URL.

**Comment by jaydeland** [comment]
@cpanato and @amyblais - this is simple an S3 routimg rule redirect to the existing releases.mattermost.com - the choice was between releases.mattermost.com/latest/<....> or latest.mattermost.com/<...> and the later seemed cleaner to me, plus we already have release.mattermost.com/server/latest as file that we didn't want to change.

If it's preffered we can look at moving it to releases.mattermost.com/latest/

**Comment by cpanato** [comment]
I leave the decision to the PM/Release manager, I just expressed my point of view.
I think the release pipeline should update the files and push to something like: `https://releases.mattermost.com/mattermost/latest/blabla-enterprise.tar.gz`

because we can have desktop following the same as well
`https://releases.mattermost.com/mattermost-desktop/latest/blabla-enterprise.tar.gz`

and

`https://releases.mattermost.com/mattermost-mmctl/latest/blabla-enterprise.tar.gz`
`https://releases.mattermost.com/mattermost-push-proxy/latest/blabla-enterprise.tar.gz`

**Comment by it33** [comment]
We used `X.X.X` to require the admin to decide what version they wanted on upgrade so they would ready the changelog and understand potentially breaking changes that could toast their system. i.e. We intentionally didn't want to point to latest. 

I just wanted to check if our thinking changed on this?

**Comment by jaydeland** [comment]
Sorry - still testing stuff - it should work now:
jpd@jpd-Oryx-Pro:~/dev/matterbuild$ wget http://latest.mattermost.com/server-linux-team
--2020-04-03 16:44:37--  http://latest.mattermost.com/server-linux-team
Resolving latest.mattermost.com (latest.mattermost.com)... 52.216.238.106
Connecting to latest.mattermost.com (latest.mattermost.com)|52.216.238.106|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://releases.mattermost.com/5.21.0/mattermost-team-5.21.0-linux-amd64.tar.gz [following]
--2020-04-03 16:44:37--  https://releases.mattermost.com/5.21.0/mattermost-team-5.21.0-linux-amd64.tar.gz
Resolving releases.mattermost.com (releases.mattermost.com)... 13.226.142.76, 13.226.142.89, 13.226.142.104, ...
Connecting to releases.mattermost.com (releases.mattermost.com)|13.226.142.76|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 162489701 (155M) [application/x-gzip]
Saving to: ‘server-linux-team’

server-linux-team                                                                  3%[=====>                                                                                                                                                                                                    ]   5.36M  2.15MB/s               ^C

**Comment by jaydeland** [comment]
@coreyhulen - do you have an opinion on latest.mattermost.com vs releases.mattermost.com/latest?

**Comment by coreyhulen** [comment]
I would lean towards releases.mattermost.com/latest, but 0/5, really no preference.

**Comment by lfbrock** [comment]
@jaydeland -- we were previously intentionally only pointing people to free E0, since it's the version people can upload a license to without needing to upgrade their server.

It could make sense to change it to show both links, but there should be a conversation about potential impacts if we do that.

**Comment by jaydeland** [comment]
@lfbrock -- so the latest URL for server should be: mattermost-enterprise-5.21.0-linux-amd64.tar.gz

**Comment by ztrayner** [comment]
**Path forward:**

- [ ] Finalize and confirm URL structure for `latest` - @jaydeland @cpanato please confirm when finalized.
- [ ] Update PR to only include E0 (remove Team reference) and use the finalized URL for `latest` - per @lfbrock 's previous post, no confirmation needed unless opinion has changed.
- [ ] Remove reference to `latest` in `upgrade.rst` and instead only reference the [version archive](https://docs.mattermost.com/administration/version-archive.html) and keep the `read the upgrade notes` warnings already in place - @it33 @jasonblais please confirm that this resolves the X.X.X concern raised by @it33.

If any of the above steps seem incorrect, please let me know.  If your confirmation has been requested, please respond.  I'm unsure of our normal convention for collaborating on something like this and if it should take place here on the PR or in [the original JIRA issue](https://mattermost.atlassian.net/browse/IS-327).  Seeing as we're all here, I'm going to post this here.

If we are truly opposed to using a `latest` endpoint, then removing all references to the [marketing download page](https://mattermost.com/download/) and replacing them with references to the [version archive](https://docs.mattermost.com/administration/version-archive.html) is also fine by me.  My purpose in this PR is to remove the circular reference from download-->docs-->download and relying on the download page as the canonical source of truth for the download string.  @jasonblais recommended against using the version archive and we collaborated and decided to use a `latest` endpoint.  I am 2/5 in favor of using a `latest` endpoint in the docs so users can have all of the information required for a new install on one page.  I am 0/5 on the format of the URL.  Note that a link to the version archive is included on each page in case the user wants to read the changelog for each release.

Thanks to everyone for collaborating on this.

**Comment by it33** [comment]
Hi @ztrayner, huge thanks for all the good intentions and work here. Can we meet with folks on this thread along with Ken, me, Corey and Lindsay to discuss the process for changes to install guides like this? 

These pages are absolutely critical to our community and company. There was a ton of testing that went into these docs, and a lot of contexts around how we do TE and EE, and how these pages integrate with the original download page:https://web.archive.org/web/20181214203140/https://mattermost.com/download/

A lot of very specific design was lost in early 2019 when we had contractors experimenting on the download page, and it's fallen out-of-sync with the doc pages--but we need to build up that understanding among the new team members so we hit the most important goals.

I spend time with Ken today talking through it and I think it's worth spending some time to get everyone on the same page.

**Comment by jaydeland** [comment]
due to the use of cloudfront with releases.mattermost.com we have no choice but to use latest.mattermost.com with this solution.

There is also a txt file being produced with the supported redirects at:
http://latest.mattermost.com

Contents of the file are:
http://latest.mattermost.com/server-linux-team
http://latest.mattermost.com/server-windows-team
http://latest.mattermost.com/mattermost-enterprise-linux
http://latest.mattermost.com/mattermost-enterprise-osx
http://latest.mattermost.com/mattermost-enterprise-windows
http://latest.mattermost.com/mattermost-desktop-amd64-deb
http://latest.mattermost.com/mattermost-desktop-i386-appimage
http://latest.mattermost.com/mattermost-desktop-i386-deb
http://latest.mattermost.com/mattermost-desktop-ia32-linux-tar
http://latest.mattermost.com/mattermost-desktop-x64-linux-tar
http://latest.mattermost.com/mattermost-desktop-x86_64-appimage
http://latest.mattermost.com/mattermost-desktop-dmg
http://latest.mattermost.com/mattermost-desktop-x64-msi
http://latest.mattermost.com/mattermost-desktop-x86-msi
http://latest.mattermost.com/mattermost-desktop-exe

**Comment by jaydeland** [comment]
Note that the latest URLs are redirect rules on the latest.mattermost.com S3 bucket that are managed by matterbuild (https://mattermost.atlassian.net/browse/IS-327)

**Comment by jaydeland** [comment]
Team Edition latest links have been added, but as @lfbrock stated we should point people to the Entreprise version to ease upgrading.

http://latest.mattermost.com/mattermost-team-linux
http://latest.mattermost.com/mattermost-team-osx
http://latest.mattermost.com/mattermost-team-windows

**Comment by kolofsen** [comment]
> http://latest.mattermost.com/mattermost-enterprise-linux
> http://latest.mattermost.com/mattermost-enterprise-osx
> http://latest.mattermost.com/mattermost-enterprise-windows

are we planning to update these three URLs to not include the word "enterprise"?

http://latest.mattermost.com/mattermost-linux
http://latest.mattermost.com/mattermost-osx
http://latest.mattermost.com/mattermost-windows

**Comment by kolofsen** [comment]
Eric Sadur and I just had a chatted about this issue and agree the intent of this PR is to remove the circular references from the docs to the /download page. As such, the next steps to resolve this issue are as @ztrayner outlined above. 

In summary:
1. Ensure the "latest" URLs for the "enterprise" downloads are as follow:
  http://latest.mattermost.com/mattermost-linux
  http://latest.mattermost.com/mattermost-osx
  http://latest.mattermost.com/mattermost-windows
2. Update the install docs to reference the "latest" URL (#1) instead of the mattermost.com/download page
3. Update the upgrade docs to reference the Version Archive instead of the mattermost.com/download page

@it33 this is the next iteration for the team to improve the install / upgrade journey. Based on the conversation you and I had earlier, we are ready to move forward without meeting with Corey and Lindsay. Can you please confirm?

**Comment by jasonblais** [comment]
I'm not against this change, but I feel strongly that this is not a small iteration. It has a wider impact on install/upgrade experience beyond removing the circular reference and deserves careful consideration, including:
 - potential risks involved in introducing `/latest` -- such as the customer example I shared on Friday. Another is the risk we introduce to community installers and other automated scripts.
 - user flows impacted (either positively or negatively), and how these compare with other self-hosted install flows like MongoDB, Elastic, GitLab, Jira Server
 - other options considered for removing the circular reference

To iterate, I'm not opposed, but I do not feel comfortable making a change in the most critical user flow for the business without careful consideration. Happy to discuss.

@kolofsen As part of the iterations involving the install/upgrade experience, please include me in the conversations in addition to Sadur, so I can provide context and use cases where appropriate.

**Comment by jaydeland** [comment]
Currently, this is what the technical piece produces - any or all can be used:

https://latest.mattermost.com/mattermost-desktop-amd64-deb
https://latest.mattermost.com/mattermost-desktop-i386-appimage
https://latest.mattermost.com/mattermost-desktop-i386-deb
https://latest.mattermost.com/mattermost-desktop-ia32-linux-tar
https://latest.mattermost.com/mattermost-desktop-x64-linux-tar
https://latest.mattermost.com/mattermost-desktop-x86_64-appimage
https://latest.mattermost.com/mattermost-desktop-dmg
https://latest.mattermost.com/mattermost-desktop-x64-msi
https://latest.mattermost.com/mattermost-desktop-x86-msi
https://latest.mattermost.com/mattermost-desktop-exe
https://latest.mattermost.com/mattermost-enterprise-linux
https://latest.mattermost.com/mattermost-enterprise-osx
https://latest.mattermost.com/mattermost-enterprise-windows
https://latest.mattermost.com/mattermost-team-linux
https://latest.mattermost.com/mattermost-team-osx
https://latest.mattermost.com/mattermost-team-windows

**Comment by jasonblais** [comment]
@justinegeffen helped gather a list of pages involving a circular reference of docs —> download page —> back to same docs page. Listing below for reference:

#### SERVER UPGRADE GUIDEs:
- https://docs.mattermost.com/administration/upgrade.html
- https://docs.mattermost.com/administration/upgrading-to-3.0.html
- https://docs.mattermost.com/administration/legacy-upgrade.html  

#### SERVER INSTALL GUIDES:
- https://docs.mattermost.com/install/install-ubuntu-1604.html#installing-mattermost-server
- https://docs.mattermost.com/install/install-ubuntu-1804.html#configuring-mattermost-server
- https://docs.mattermost.com/install/install-debian.html#installing-mattermost-server
- https://docs.mattermost.com/install/install-rhel-6.html#installing-mattermost-server
- https://docs.mattermost.com/install/install-rhel-7.html#installing-mattermost-server

#### DESKTOP APP INSTALL GUIDES:
- https://docs.mattermost.com/install/desktop.html

Below are other pages we direct the reader to the download page for the latest client or server (which is fine for now in the existing journey):
- https://docs.mattermost.com/overview/product.html#mattermost-editions
- https://docs.mattermost.com/developer/slash-commands.html
- https://docs.mattermost.com/administration/hipchat-migration-guidelines.html
- https://docs.mattermost.com/getting-started/welcome_email.html
- https://docs.mattermost.com/help/apps/desktop-changelog.html

----

Moreover, when reviewing current download a/b test metrics, we noticed it appears the circular reference of docs > download > docs does not actually often happy.

However, there was some noise in the metrics (due to a redirect in the a/b test instrumentation), so we'd want to take a closer look prior to the start of the a/b test.

----

@ztrayner @esadur As next steps, I'd propose we:

1. Look at the occurrence of people going from docs > download page > docs, so we can estimate the impact and hence the priority
2. With info from 1, prioritize against other initiatives Eric has in the customer journey - which includes those in the install/upgrade experience and the download page we've discussed
3. If the circular reference is deemed highest priority, then we should set up a way to measure the impact of this change, so we can identify negative effects to the experience, if any (the change could be a hard-coded server link, use of the `/latest` URL, or something else).

Let me know if there are any questions or concerns. We can discuss more on our next sync-up call on Thursday.

**Pull request closed without merging**